src/Driver/Driver.jl

```diff
@@ -183,115 +190,8 @@ function init(; disable_gpu=false)
     return nothing
 end
 
-"""
-    CLIMA.SolverConfiguration
-
-Parameters needed by `CLIMA.solve!()` to run a simulation.
-"""
-struct SolverConfiguration{FT}
-    name::String
-    mpicomm::MPI.Comm
-    dg::DGModel
-    Q::MPIStateArray
-    t0::FT
-    timeend::FT
-    dt::FT
-    init_on_cpu::Bool
-    numberofsteps::Int
-    init_args
-    solver
-end
-
-"""
-    DGmethods.courant(local_cfl, solver_config::SolverConfiguration;
-                      Q=solver_config.Q, dt=solver_config.dt)
-
-Returns the maximum of the evaluation of the function `local_courant`
-pointwise throughout the domain with the model defined by `solver_config`. The
-keyword arguments `Q` and `dt` can be used to call the courant method with a
-different state `Q` or time step `dt` than are defined in `solver_config`.
-"""
-DGmethods.courant(f, sc::SolverConfiguration; Q=sc.Q, dt = sc.dt) =
-  DGmethods.courant(f, sc.dg, sc.dg.balancelaw, Q, dt)
-
-"""
-    CLIMA.setup_solver(t0, timeend, driver_config)
-
-Set up the DG model per the specified driver configuration and set up the ODE solver.
-"""
-function setup_solver(t0::FT, timeend::FT,
-                      driver_config::DriverConfiguration,
-                      init_args...;
-                      init_on_cpu=false,
-                      ode_solver_type=nothing,
-                      ode_dt=nothing,
-                      modeldata=nothing,
-                      Courant_number=0.4,
-                      diffdir=EveryDirection(),
-                     ) where {FT<:AbstractFloat}
-    @tic setup_solver
-
-    bl = driver_config.bl
-    grid = driver_config.grid
-    numfluxnondiff = driver_config.numfluxnondiff
-    numfluxdiff = driver_config.numfluxdiff
-    gradnumflux = driver_config.gradnumflux
-
-    # create DG model, initialize ODE state
-    dg = DGModel(bl, grid, numfluxnondiff, numfluxdiff, gradnumflux,
-                 modeldata=modeldata,
-                 diffusion_direction=diffdir)
-    @info @sprintf("Initializing %s", driver_config.name)
-    Q = init_ode_state(dg, FT(0), init_args...; init_on_cpu=init_on_cpu)
-
-    # TODO: using `update_aux!()` to apply filters to the state variables and
-    # `update_aux_diffusive!()` to calculate the vertical component of velocity
-    # using the integral kernels -- these should have their own interface
-    update_aux!(dg, bl, Q, FT(0))
-    update_aux_diffusive!(dg, bl, Q, FT(0))
-
```

Comment:
My guess is that this is the problem. You dropped the update_aux.

Reply:
Thank you!